authorization-server/src/modules/auth/guards/valid-refresh-token.guard.ts

```diff
@@ -27,10 +63,31 @@ export class ValidRefreshTokenGuard implements CanActivate {
       refreshToken = body?.refreshToken;
     }
 
-    if (!refreshToken) {
-      return false;
+    return refreshToken;
+  }
+
+  extractFromQueryStringOrCookies(
+    cookies?: Record<string, string>,
+    query?: { refresh_token?: string },
+  ): string | undefined {
+    let refreshToken: string;
+    const AUTH_HEADER_ENABLED = this.config.get<boolean>('AUTH_HEADER_ENABLED');
+    const AUTH_COOKIE_ENABLED = this.config.get<boolean>('AUTH_COOKIE_ENABLED');
+
+    const refreshTokenFromQueryString: string = query && query['refresh_token'];
```

Comment:
I would have thought that the type here is `string | undefined`. I'm surprised typescript doesn't give an error  .
```suggestion
    const refreshTokenFromQueryString: string | undefined = query && query['refresh_token'];
```